src/main/java/swingtree/UIForTabbedPane.java

```diff
@@ -442,6 +446,68 @@ public final UIForTabbedPane<P> add( Tab tab )
         ._this();
     }
 
+    /**
+     * Dynamically generates tabs for items in a {@link Vals} list and automatically updates them
+     * when the items change. The items are typically view model instances, but can be any type.
+     * <p>
+     * The provided {@link TabSupplier} lambda is invoked with each item from the list,
+     * returning a {@link Tab} to be added to the {@link JTabbedPane} wrapped by this builder.
+     *
+     * <p><b>Note:</b> Binding tabs to a {@link Vals} list assumes no other tabs are present.
+     *
+     * @param <M>         The type of items in the {@link Vals} list.
+     * @param tabModels   A list of items, typically view model instances.
+     * @param tabSupplier A lambda to generate a {@link Tab} for each item.
+     * @return This instance, allowing for builder-style method chaining.
+     */
+    public <M> UIForTabbedPane<P> add(Vals<M> tabModels, TabSupplier<M> tabSupplier) {
+        return _with(thisComponent -> {
+            _onShow(tabModels, thisComponent, (p, delegate) -> {
+                Vals<M> newValues = delegate.newValues();
+                Vals<M> oldValues = delegate.oldValues();
+
+                switch (delegate.changeType()) {
+                    case SET:
+                        for (int i = 0; i < newValues.size(); i++) {
+                            int position = delegate.index() + i;
+                            _updateTabAt(position, newValues.at(i).get(), tabSupplier, p);
```

Comment:
Thanks for finding this. `.get()` was a bug. Actually I support `null`. If a model is `null`, a special tab is inserted, the same if an error occurs while creating a tab.

But maybe we want to change this behavior?

```Java
    private static final Tab TAB_ERROR = UI.tab("Error Tab");
    private static final Tab TAB_NULL = UI.tab("Empty Tab");
```

Reply:
No I really like the approach of an error and a null tab.
In order for binding to work consistently we absolutely need to have something there.
If we alternatively just not add a null or error tab any updates comming from the properties afterwards
may target the wrong tabs! That would break in all sorts if unpredictable ways.

And just throwing an exception here is also nonsensical, since that has the same implications
as above, and it then may also break the GUI in even more ways.